addons/portal/controllers/portal_thread.py

```diff
@@ -44,45 +25,3 @@ def portal_chatter_init(self, thread_model, thread_id, **kwargs):
         if request.env.user.has_group("website.group_website_restricted_editor"):
             store.add(partner, {"is_user_publisher": True})
         return store.get_result()
-
-    @http.route('/mail/chatter_fetch', type='jsonrpc', auth='public', website=True)
-    def portal_message_fetch(self, thread_model, thread_id, fetch_params=None, **kw):
-        # Only search into website_message_ids, so apply the same domain to perform only one search
-        # extract domain from the 'website_message_ids' field
-        model = request.env[thread_model]
-        field = model._fields['website_message_ids']
-        domain = Domain.AND([
-            self._setup_portal_message_fetch_extra_domain(kw),
-            field.get_comodel_domain(model),
-            [('res_id', '=', thread_id), '|', ('body', '!=', ''), ('attachment_ids', '!=', False),
-             ("subtype_id", "=", request.env.ref("mail.mt_comment").id)]
-        ])
-
-        # Check access
-        Message = request.env['mail.message']
-        if kw.get('token'):
-            access_as_sudo = ThreadController._get_thread_with_access(
-                thread_model, thread_id, token=kw.get("token"),
-            )
-            if not access_as_sudo:  # if token is not correct, raise Forbidden
-                raise Forbidden()
-            # Non-employee see only messages with not internal subtype (aka, no internal logs)
-            if not request.env.user._is_internal():
-                domain = Domain.AND([Message._get_search_domain_share(), domain])
-            Message = request.env["mail.message"].sudo()
-        res = Message._message_fetch(domain, **(fetch_params or {}))
-        messages = res.pop("messages")
-        return {
-            **res,
-            "data": {"mail.message": messages.portal_message_format(options=kw)},
-            "messages": messages.ids,
-        }
-
-    def _setup_portal_message_fetch_extra_domain(self, data):
-        return []
-
-    @http.route(['/mail/update_is_internal'], type='jsonrpc', auth="user", website=True)
```

Comment:
Is it dead code? Didn't we kill it my mistake with previous PR?

Reply:
We used to have a button that could change the visibility of a message in the portal. We don't do that anymore.

Reply:
Was it removed on purpose? Why PR did it?

Reply:
I think it's important especially on product page and slides to have moderation (ability to remove unwanted messages from public view).

Reply:
Yes, it was requested by PO when the portal chatter was converted and it was done on purpose. We can bring it back if we want though.

Reply:
If the message has no content (and no rating), we can prevent it from being shown on the public page. In fact, I'm doing this in another WIP PR for the modules that have a rating feature, to not show deleted messages. It could be (if we want) for all deleted messages in the portal (with or without rating).